### PR TITLE
[TACHYON-787] add test for BufferUtils#cleanDirectBuffer

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-787

As suggested by @apc999 , allocate & free 16 MB of memory for 1024 times, make sure there is no OOM

### CODE REVIEW

File: common/src/test/java/tachyon/util/io/BufferUtilsTest.java [hunk 1]
```diff
@@ -44,4 +44,18 @@ public void cloneDirectByteBufferTest() {
...
   }
-}
+
+  @Test
+  public void cleanDirectBufferTest() {
+    final int MAX_ITERATIONS = 1024;
+    try {
+      for (int i = 0; i < MAX_ITERATIONS; ++i) {
+        final int bufferSize = 16 * 1024 * 1024;
+        ByteBuffer buf = ByteBuffer.allocate(bufferSize);
```
Comment (calvinjia) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: This needs to be a direct byte buffer I think.

Comment (miasantreble) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: good catch! fixed. Actually I'm amazed that Jenkins found out the bug as well

**Comment by apc999** [comment]
@miasantreble , if you comment out  `cleanDirectBuffer`, do you see test failure ?

**Comment by miasantreble** [comment]
@apc999 with the addition of `bufferArray` to keep a reference to each buffer allocated, now I do see test failure when `cleanDirectBuffer` is commented out, thanks!

File: common/src/test/java/tachyon/util/io/BufferUtilsTest.java [hunk 2]
```diff
@@ -44,4 +44,21 @@ public void cloneDirectByteBufferTest() {
...
-}
+
+  @Test
+  public void cleanDirectBufferTest() {
+    final int MAX_ITERATIONS = 1024;
+    final int BUFFER_SIZE = 16 * 1024 * 1024;
+    // bufferArray keeps reference to each buffer to avoid auto GC
+    ByteBuffer[] bufferArray = new ByteBuffer[MAX_ITERATIONS];
+    try {
+      for (int i = 0; i < MAX_ITERATIONS; ++i) {
```
Comment (apc999) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: `++i` -> `i ++` or `++ i` to keep consistent Tachyon code style.

**Comment by apc999** [comment]
Very glad to know that `cleanDirectBuffer` works as expected! Thanks for the patch!
I left one style comment. After that let's merge this PR.

File: common/src/test/java/tachyon/util/io/BufferUtilsTest.java [hunk 3]
```diff
@@ -44,4 +44,21 @@ public void cloneDirectByteBufferTest() {
     bufClone = BufferUtils.cloneByteBuffer(bufDirect);
     Assert.assertTrue(bufClone.equals(bufDirect));
   }
-}
+
+  @Test
+  public void cleanDirectBufferTest() {
```
Comment (apc999) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: Can you javadoc a bit what's going on in this test, and what's expected.

File: common/src/test/java/tachyon/util/io/BufferUtilsTest.java (hunk 2 above)
Comment (miasantreble) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: done

File: common/src/test/java/tachyon/util/io/BufferUtilsTest.java (hunk 3 above)
Comment (miasantreble) [inline common/src/test/java/tachyon/util/io/BufferUtilsTest.java]: done

**Comment by apc999** [comment]
+1 LGTM.

Thanks for the patch!

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**